src/Data/Text/Internal.hs

```diff
@@ -252,6 +252,6 @@ pack xs = runST $ do
         go (off + d) cs
   len <- go 0 xs
   arr <- A.unsafeFreeze marr
-  return (Text arr 0 len)
+  return (text arr 0 len)
```

Comment:
It would be better to add a case `pack [] = empty`, skipping the `A.new` altogether.

Reply:
Done in 8a60cdf06a7cc4fab1c4e907c8da22979672867e